### ISSUE TITLE
Bug: findComponent doesn't find anything when using @vue/compat

### ISSUE THREAD

**Issue by tom751** [vuejs/test-utils#2385]
**Describe the bug**
I'm currently migrating a large Vue 2.7 app over to Vue 3 using the `@vue/compat` build. The apps work but the tests always fail because  `findComponent` or `findAllComponents` never seem to find anything when passing in a component. For example, we have a lot of this type of expectation in our tests

```ts
expect(wrapper.findComponent(NoticeBanner).exists()).toBe(true)
```

If I log out the html of the wrapper with `wrapper.html()`, we can see that the component is there:
```html
<main>
  <noticebanner msg="Hello world"></noticebanner>
</main>
```

If I add a class to the component, and query that with `.find('.some-class')`, then that works - but due to the number of component tests I have, it's not really feasible for me to update hundreds of files.

The problem definitely seems to be with using the compat build, because if I remove the alias from the Vite config (`vue: '@vue/compat'`) then `findComponent` works again.

**To Reproduce**
Here's an example repo showing the issue https://github.com/tom751/vtu-compat-issue - this was a new project created with `npm create vue@legacy` and then I followed the migration guide to update to Vue 3 with the compat build (see the commits).

- run `npm run test:unit` - HomeView.spec.ts will fail
- comment out `vue: '@vue/compat'` in vite.config.ts and rerun the tests - they now pass

**Expected behavior**
`findComponent/findAllComponents` should find the component when it exists and passing it as an argument

**Additional context**
Removing the alias to `@vue/compat` causes `findComponent` to work again, but then tests start to fail on Vue 2 syntax that hasn't been migrated yet in the codebase.

I've also tried various versions of vue/test-utils/compat and the behaviour is the same.

**Comment by chrismooreajbell** [comment]
Hi there,

I've had a look at this. The test utils package contains tests against this scenario which show that your example should work.

I checked out your repo and noticed you are using vitest as your test runner; the vue test utils package uses jest.

Since I had your repo downloaded I upgraded all your vite/vitest dependencies to latest and added the following entry into your vite config, in the test option:
```
server: {
  deps: {
    inline: ['@vue/test-utils']
  },
},
```

I'm not entirely sure why this works (I _think_ it forces vitest to use the cjs build of vue test utils) and it appears to resolve your problem. See https://github.com/vitest-dev/vitest/issues/1886#issuecomment-1221890492 for more info.

This may result in creating an issue against vitest but I'm unsure - interested to hear if the library owners might chip in.
